### PR TITLE
48572 Split out admin functionality into separate web app

### PR DESCRIPTION
**Pull request by Avinash15515**

### CODE REVIEW

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-﻿using Microsoft.EntityFrameworkCore;
+﻿using System.Diagnostics.CodeAnalysis;
```
[the remark this answers is not in the document]
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs:1]: Removed

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            {
+                loggingBuilder.AddConfiguration(configuration.GetSection("Logging"));
+                loggingBuilder.AddConsole();
+                loggingBuilder.AddDebug();
+                loggingBuilder.AddAzureWebAppDiagnostics();
+            });
+            services.Configure<EventHubLoggingConfiguration>(configuration.GetSection("EventHubLoggingConfiguration"));
+            services.Configure<RadioNavigationalWarningConfiguration>(configuration.GetSection("RadioNavigationalWarningConfiguration"));     
+            services.Configure<FileShareServiceConfiguration>(configuration.GetSection("FileShareService"));
+            services.Configure<RadioNavigationalWarningsContextConfiguration>(configuration.GetSection("RadioNavigationalWarningsContext"));
```
[the remark this answers is not in the document]
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Startup.cs]: Removed this configuration completely.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Startup.cs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+            var msiDBConfiguration = new RadioNavigationalWarningsContextConfiguration();
+            configuration.Bind("RadioNavigationalWarningsAdminContext", msiDBConfiguration);
+            services.AddDbContext<RadioNavigationalWarningsContext>(options => options.UseSqlServer(msiDBConfiguration.ConnectionString));
+
+            services.AddScoped<IEventHubLoggingHealthClient, EventHubLoggingHealthClient>();
+            services.AddScoped<IAuthFssTokenProvider, AuthFssTokenProvider>();
```
[the remark this answers is not in the document]
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Startup.cs]: Removed

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Views/Shared/_Layout.cshtml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                    </div>
+                    <p class="footer-text">&copy; Crown copyright @DateTime.Now.Year UK Hydrographic Office</p>
+                </div>
+            </div>
+        </div>
+    </footer>
+    <script src="~/lib/jquery/dist/jquery.min.js"></script>
+    <script src="~/lib/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
+    <script src="~/js/site.js" asp-append-version="true"></script>
+    <script src="~/js/Custom/RadioNavigationalWarnings-admin.js" asp-append-version="true"></script>
```
[the remark this answers is not in the document]
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Views/Shared/_Layout.cshtml:48]: Removed reference to Jquery and Admin.Js from _Layout.cshtml of MSI Public App

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Views/Shared/_Layout.cshtml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                        <a class="col" href="#" target="_blank" tabindex="0" id="accessibilitylink" aria-label="Accessibility opens in new tab">Accessibility</a>
+                    </div>
+                    <p class="footer-text">&copy; Crown copyright @DateTime.Now.Year UK Hydrographic Office</p>
+                </div>
+            </div>
+        </div>
+    </footer>
+    <script src="~/lib/jquery/dist/jquery.min.js"></script>
+    <script src="~/lib/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
+    <script src="~/js/site.js" asp-append-version="true"></script>
```
[the remark this answers is not in the document]
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Views/Shared/_Layout.cshtml:47]: Removed

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Views/Shared/_MSILayout.cshtml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    <div class="row">
+        <div class="col-md-2 msi-tabs-container">
+            <!--left nav-->
+            @if (ViewContext.RouteData.Values["controller"]?.ToString() == "NoticesToMariners")
+            {
+                <partial name="_NMLeftMenu"></partial>
+            }
+            else if (ViewContext.RouteData.Values["controller"]?.ToString() == "RadioNavigationalWarnings")
+            {
+                <partial name="_RNWLeftMenu"></partial>
```
[the remark this answers is not in the document]
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Views/Shared/_MSILayout.cshtml]: Removed

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Views/Shared/_RNWLeftMenu.cshtml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+﻿<ul class="nav flex-column msi-page-nav">
+    <li class="nav-item">
+        <a href="@Url.Action("Index", "RadioNavigationalWarnings")"
+           class="@(ViewContext.RouteData.Values["Action"].ToString() == "Index" ? "nav-link active" : "nav-link")"
```
[the remark this answers is not in the document]
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Views/Shared/_RNWLeftMenu.cshtml:4]: Removed

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/appsettings.json [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    "KeyVaultSettings": {
+        "ServiceUri": ""
+    },
+    "FileShareService": {
+        "BaseUrl": "",
+        "BusinessUnit": "",
+        "ProductType": "",
+        "FssClientId": "",
+        "PageSize": "100",
+        "Start": "0"
```
[the remark this answers is not in the document]
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/appsettings.json]: Removed

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/appsettings.json [hunk 9]
```diff
@@ -0,0 +1,46 @@
+{
+    "Logging": {
+        "PathFormat": "Logs/UKHO.MaritimeSafetyInformation.Web-{Date}.txt",
```
[the remark this answers is not in the document]
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/appsettings.json]: Removed

**Review by usha13529 — APPROVED** [review]
Thanks for the changes @Avinash15515. Looks good to me.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/appsettings.Development.json [hunk 10]
```diff
@@ -0,0 +1,15 @@
+{
+    "Logging": {
+        "LogLevel": {
+            "Default": "Information",
+            "Microsoft.AspNetCore": "Warning"
+        }
+    },
+
```
Comment (barrie-cooper) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/appsettings.Development.json:8]: Please remove blank line
```suggestion
```

**Review by barrie-cooper — COMMENTED** [review]
A couple of minor changes plus:

1. I get the following build errors which need to be addressed please:

   image

1. In the `UKHO.MaritimeSafetyInformation.Web` project `appsettings.json` & `appsettings.Development.json` both have the following admin setting still in it:
   ```
       "RadioNavigationalWarningConfiguration": {
           "AdminListRecordPerPage":  ""
       },
   ```

File: azure-pipelines.yml [hunk 11]
```diff
@@ -388,6 +405,8 @@ stages:
         variables:
           - name: WEB_APP_NAME
             value: $[ dependencies.QADeployTerraform.outputs['QADeployTerraform.TerraformDeploy.WEB_APP_NAME']  ]
+          - name: ADMIN_WEB_APP_NAME
+            value: $[ dependencies.DevDeployTerraform.outputs['DevDeployTerraform.TerraformDeploy.ADMIN_WEB_APP_NAME']  ]
```
[the remark this answers is not in the document]
Comment (Avinash15515) [reply azure-pipelines.yml]: Changes done by Tejasi

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/appsettings.Development.json (hunk 10 above)
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/appsettings.Development.json]: Removed blank line

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Views/Shared/_RNWLeftMenu.cshtml [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+        </a>
+    </li>
+    <li class="nav-item">
+        <a href="@Url.Action("About", "RadioNavigationalWarnings")"
+           class="@(ViewContext.RouteData.Values["Action"].ToString() == "About" ? "nav-link active" : "nav-link")"
+           aria-current="@(ViewContext.RouteData.Values["Action"].ToString() == "About" ? "true" : "")">
+            About
+        </a>
+    </li>
+</ul>
```
[the remark this answers is not in the document]
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Views/Shared/_RNWLeftMenu.cshtml]: File deleted

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web.csproj [hunk 13]
```diff
@@ -16,6 +16,10 @@
     <None Remove="Logs\**" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Compile Remove="Controllers\RadioNavigationalWarningsAdminController.cs" />
+  </ItemGroup>
+
```
[the remark this answers is not in the document]
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web.csproj]: Section removed

**Review by barrie-cooper — COMMENTED** [review]
Sorry @Avinash15515, just one tiny change then I'm happy.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/UKHO.MaritimeSafetyInformationAdmin.Web.csproj [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+    <Compile Include="..\UKHO.MaritimeSafetyInformation.Web\Services\Interfaces\IRNWService.cs" Link="Services\Interfaces\IRNWService.cs" />
+    <Compile Include="..\UKHO.MaritimeSafetyInformation.Web\Services\RNWRepository.cs" Link="Services\RNWRepository.cs" />
+    <Compile Include="..\UKHO.MaritimeSafetyInformation.Web\Services\RNWService.cs" Link="Services\RNWService.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Folder Include="Filters\" />
+    <Folder Include="Services\Interfaces\" />
+  </ItemGroup>
+
```
[the remark this answers is not in the document]
Comment (usha13529) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/UKHO.MaritimeSafetyInformationAdmin.Web.csproj]: thanks for highlighting this. Section removed.

**Pull request merged**